src/features/Configuration/boardDefinitions.ts

```diff
@@ -142,10 +151,27 @@ export function generatePortMap(
 ): Map<number, PmicPortDescription> {
     const pinMap = new Map<number, PmicPortDescription>();
 
+    // TODO: check all these isArray() stuff; looks dirty
     boardControllerConfigDefinition?.pmicPorts?.forEach(port => {
-        if (port.portId) {
-            pinMap.set(port.port, { id: port.portId });
+        if (!port.portId) {
+            return;
+        }
+
+        if (Array.isArray(port.port) && Array.isArray(port.portId)) {
+            port.port.forEach((p, idx) => {
+                // @ts-expect-error TODO: check why it says port.portId can be undefined
+                pinMap.set(p, { id: port.portId[idx] });
+            });
+
+            return;
```

Comment:
// @ts-expect-error TODO: check why it says port.portId can be undefined
This is a Typescript issue as check is out of lambda and is in side an object and prop is not a const 

consider 
```suggestion
        const portId = port.portId
        if (!portId) {
            return;
        }

        if (Array.isArray(port.port) && Array.isArray(port.portId)) {
            port.port.forEach((p, idx) => {
                pinMap.set(p, { id: portId[idx] });
            });

            return;
```